swift-generator/src/main/resources/templates/java/common.st

```diff
@@ -333,3 +336,47 @@ public String toString()
         .toString();
 }
 >>
+
+_union_constructor(context) ::= <<
+
+private Object value;
+private int id = -1;
+private String name;
+
+public <context.javaName>() {
```

Comment:
When building, you only call setSomeField() once. For example, we don't want users to call builder.setSomeField().setAnotherField(). This condition is upheld below in the setters.

You're right, we should remove the condition check in the private constructor.